osu.Game/Beatmaps/BeatmapManager.cs

```diff
@@ -442,5 +449,53 @@ public void Dispose()
         }
 
         #endregion
+
+        /// <summary>
+        /// Import a beatmap directly from game resources.
+        /// </summary>
+        /// <remarks>
+        /// Will mark as protected.
+        /// </remarks>
+        /// <param name="beatmapFile"></param>
+        /// <param name="beatmapHash"></param>
+        /// <returns></returns>
+        public WorkingBeatmap? ImportFromGameResources(string beatmapFile, string beatmapHash)
+        {
+            WorkingBeatmap? beatmap = null;
+
+            // First check if the beatmap is already imported and in a good state.
+            var setInfo = QueryBeatmapSet(b => b.Protected && b.Hash == beatmapHash);
+
+            if (setInfo != null)
+            {
+                setInfo.PerformRead(s =>
+                {
+                    if (s.Beatmaps.Count == 0)
+                        return;
+
+                    beatmap = GetWorkingBeatmap(s.Beatmaps.First());
+                });
+
+                if (beatmap != null)
+                    return beatmap;
+            }
+
+            // Intro beatmaps are generally made using the osu! ruleset.
+            // It might not be present in test projects for other rulesets.
+            // To avoid import errors / failures, skip import if we don't have an osu! ruleset.
+            if (rulesets.GetRuleset(0) == null)
+                return null;
```

Comment:
Not sure how valid this is as you could potentially have a tutorial beatmap for another ruleset, maybe better moving it back to `IntroScreen`?

Reply:
Not sure about this one. The idea is that the code before this call should still run (I think). Will do a second pass and see if it can be improved at all.